static/elements/chromedash-featurelist.html

```diff
@@ -130,38 +178,76 @@
           }
           this.filtered = this.features;
         } else {
+
           val = val.trim();
           if (history && history.replaceState) {
             history.replaceState({id: null}, document.title, '/features#' + val);
           }
 
-          // owner: user@chromium.org
-          var ownerMatch = val.match(/^owner:\s*(.*)/);
-          // category: Multimedia
-          var categoryMatch = val.match(/^category:\s*(.*)/);
-          // Returns operator and version query e.g. ["<=25", "<=", "25"].
-          var operatorMatch = /^([<>=]=?)\s*([0-9]+)/.exec(val);
-
-          if (ownerMatch) {
-            this.filtered = this._filterOnOwner(ownerMatch[1]);
-          } else if (categoryMatch) {
-            this.filtered = this._filterOnCategory(categoryMatch[1]);
-          } else if (operatorMatch) {
-            this.filtered = this._filterOnOperation(operatorMatch[1], operatorMatch[2]);
+          // Match words separated by whitespace and/or ":" and/or ordered
+          // comparison operator.
+          var regExp = /("([^"]+)"|([^:<>= \f\n\r\t\v\u00a0\u1680\u180e\u2000-\u200a\u2028\u2029\u202f\u205f\u3000\ufeff]+))([:<>= \f\n\r\t\v\u00a0\u1680\u180e\u2000-\u200a\u2028\u2029\u202f\u205f\u3000\ufeff]+)?/g;
+          // Array of matches, each an array of the form:
+          // [ full match, quoted-string-or-word, contents-of-quoted-string,
+          //   word, separator ].
+          var reMatches = [];
+          var match;
+          while ((match = regExp.exec(val)) !== null) {
+            reMatches.push(match);
+          }
+
+          // Query parts of the form: "name : value".
+          var propertyQueries = [];
+          // Query parts of the form: "name = value", "name < value", etc.
+          var operatorQueries = [];
+          // Other words in the query (not matching forms above).
+          var keywordQueries = [];
+
+          // Accumulate keyword and property queries.
+          var i;
+          for (i = 0; i < reMatches.length; i++) {
```

Comment:
Switched to `function(...) { ... for (var i...) { ... } ... for (var i...) { ... } ... }` (harmless variable redeclaration) as per discussion.